src/crypto.ts

```diff
@@ -190,6 +192,49 @@ function valid(signingPublicKey: string) {
   return _internalValid
 }
 
+/**
+ * Helper function to encrypt file with a unique keypair for each submission.
+ * @param blob The file to encrypt
+ * @param formPublicKey The base-64 encoded public key
+ * @returns Promise holding the encrypted file
+ * @throws error if any of the encrypt methods fail
+ */
+async function encryptFile (blob: Blob, formPublicKey: string): Promise<EncryptedFileContent> {
+  const binary = new Uint8Array(await blob.arrayBuffer())
+  const submissionKeypair = generate()
+  const nonce = nacl.randomBytes(24)
+  return {
+    submissionPublicKey: submissionKeypair.publicKey,
+    nonce: encodeBase64(nonce),
+    blob: new Blob([nacl.box(
+      binary,
+      nonce,
+      decodeBase64(formPublicKey),
+      decodeBase64(submissionKeypair.secretKey)
+    )])
+  }
+}
+
+/**
+ * Helper function to decrypt a file
+ * @param formSecretKey Secret key as a base-64 string
+ * @param encrypted Object returned from encryptFile function
+ * @param encrypted.submissionPublicKey The submission public key as a base-64 string
+ * @param encrypted.nonce The nonce as a base-64 string
+ * @param blob The encrypted file as a Blob object
+ */
+async function decryptFile (formSecretKey: string, { submissionPublicKey, nonce, blob }: EncryptedFileContent ): Promise<Blob> {
+  const encryptedBinary = new Uint8Array(await blob.arrayBuffer())
+  const decryptedBinary = nacl.box.open(
+    encryptedBinary,
+    decodeBase64(nonce),
+    decodeBase64(submissionPublicKey),
+    decodeBase64(formSecretKey),
+  )
+  if (decryptedBinary) return new Blob([decryptedBinary])
+  else throw new Error('Decryption Error')
```

Comment:
Do we want to follow the convention of `decrypt` where null is returned instead of any errors being thrown?

We should stick to one or the other. Both are fine, but I am actually in favour of throwing so that it is easier to debug rather than wondering why `decrypt` is returning `null`.

Reply:
This is an async function, so it returns a rejected Promise with an error instead. Would it be strange if we threw an empty Error object?

Reply:
If we follow `decrypt`, then yes a `null` wrapped inside a Promise will be returned, haha.

But if we want to throw errors, then `decrypt` should also return an error instead of null.